hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java

```diff
@@ -112,8 +113,19 @@ public ServerConnection get(Address address, int streamId) {
     }
 
     public Set<Address> getKnownAliases(TcpServerConnection connection) {
-        Plane plane = planes[connection.getPlaneIndex()];
-        return plane.getAddresses(connection);
+        Plane plane;
+        if (connection.getPlaneIndex() == -1) {
+            // It could be the case that planeIndex of the connection may not
+            // be set. In this case, we return only the remote address of
+            // connection. Since the plane index is set before the connection
+            // registration (handshake processing), we don't expect this
+            // connection exist in connectionMaps of the planes and don't
+            // search on them.
+            return Collections.singleton(connection.getRemoteAddress());
```

Comment:
Can this be a singleton with `null` as single element? Will it be handled gracefully by consumers of `getKnownAliases`?

Reply:
This address is not set in its initialization and I think I should behave it as it can be null. Thanks for pointing this out